### PR TITLE
feat: add skipAudit, dryRun, and import strategy to UI [DHIS2-17489]

### PR DESCRIPTION
**Pull request by tomzemp**
## Background
See https://dhis2.atlassian.net/browse/DHIS2-17489

This PR updates to 
- provide Advanced options (can be toggled to show from exchange main form) to set skipAudit, dryRun, and importStrategy

image

- skip audit requires the `F_SKIP_DATA_IMPORT_AUDIT` authority, and we had a long discussion about whether a user should be warned about creating an exchange with skipAudit:true if they do not have that authority. Since the backend allows this and since many exchanges are executed as jobs where they will be run by the system user (who will have the authority), it was decided not to warn the user at time of exchange creation.
- at time of submit from within the app, there is a warning in the submit modal (if skipAudit:true and the exchange is internal and the user does not have skip audit authority):
image

I've also refactored some of the exchange form lightly to be able to keep the advanced options in a separate file.

### Testing
I have done manual testing and not noticed anything in particular.

I have added automated tests:
- in addItem.test.js that covers the basic functionality (test that it shows based on version, that it sends the right values in the POST, etc.)
- tests for warnings displays (submitModal.test.js and advanced-options.test.js)

I have not added automated tests for editing, because the UI currently requires you to reupdate authentication details. But this was due to a bug on the backend that is now fixed (https://dhis2.atlassian.net/issues/DHIS2-16440), so we should probably update the UX here to stop you from needing to reenter the authentication details.

### CODE REVIEW

File: src/components/edit/exchange-update/advanced-options.module.css [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* move if using */
+.prefix {
+    font-family: monospace;
+    font-size: 13px;
+    color: var(--colors-grey800);
+    background: var(--colors-grey300);
+    padding: 2px var(--spacers-dp4);
+    margin-right: var(--spacers-dp4);
```
Comment (Birkbjo) [inline src/components/edit/exchange-update/advanced-options.module.css:8]:
```suggestion
    padding-block: 2px;
    padding-inline: var(--spacers-dp4);
    margin-inline-start: var(--spacers-dp4);
```

File: src/components/edit/exchange-update/external-edit-warning.module.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.editWarningButton {
+    margin: var(--spacers-dp8) 0 0 0;
```
Comment (Birkbjo) [inline src/components/edit/exchange-update/external-edit-warning.module.css:2]: Think this needs `margin-block` and `margin-inline`?

File: src/context/feature-toggle-context/feature-toggle-provider.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ const FeatureToggleProvider = ({ children }) => {
 
     const providerValue = {
         outputDataItemIdSchemeAvailable: apiVersion >= 41,
+        // skipAuditDryRunImportStrategyAvailable: apiVersion >= 41,
+        skipAuditDryRunImportStrategyAvailable: true,
```
Comment (Birkbjo) [inline src/context/feature-toggle-context/feature-toggle-provider.js:12]: Is this intended?

**Review by Birkbjo — APPROVED** [review]
LGTM, some minor comments.

File: src/context/feature-toggle-context/feature-toggle-provider.js (hunk 3 above)
Comment (tomzemp) [reply src/context/feature-toggle-context/feature-toggle-provider.js]: Yikes 😱. No, not intentional. Was trying to figure out something with a test that was passing incorrectly. I've fixed this (and the test) now. 🙏

**Pull request merged**